include/svs/index/vamana/multi.h

```diff
@@ -0,0 +1,563 @@
+    using search_parameters_type = typename ParentIndex::search_parameters_type;
+    using external_id_type = typename ParentIndex::external_id_type;
+    using scratchspace_type = typename ParentIndex::scratchspace_type;
+    using distance_type = Dist;
+    using label_type = size_t;
+    using graph_type = Graph;
+    using data_type = Data;
+
+  private:
+    distance_type distance_;
+    external_id_type counter_{0};
+    std::unique_ptr<ParentIndex> index_{nullptr};
+    std::unordered_map<label_type, std::vector<external_id_type>> label_to_external_;
+    std::unordered_map<external_id_type, label_type> external_to_label_;
+
+    template <class Labels>
+    void
+    prepare_added_id_by_label(const Labels& labels, std::vector<external_id_type>& adds) {
+        for (const auto l : labels) {
+            if (label_to_external_.find(l) == label_to_external_.end()) {
+                label_to_external_.insert({l, std::vector<external_id_type>{}});
+            }
+
+            size_t new_external_id = counter_++;
+            label_to_external_[l].push_back(new_external_id);
+            external_to_label_.insert({new_external_id, l});
+            adds.push_back(new_external_id);
+        }
+    }
+
+  public:
+    template <typename Labels, typename ThreadPoolProto>
+    MultiMutableVamanaIndex(
+        const VamanaBuildParameters& parameters,
+        Data data,
+        const Labels& labels,
+        Dist distance_function,
+        ThreadPoolProto threadpool_proto,
+        svs::logging::logger_ptr logger = svs::logging::get()
+    )
+        : distance_(std::move(distance_function)) {
+        std::vector<external_id_type> adds;
+        adds.reserve(labels.size());
+        prepare_added_id_by_label(labels, adds);
+        index_ = std::make_unique<ParentIndex>(
+            parameters,
+            std::move(data),
+            std::move(adds),
+            distance_,
+            std::move(threadpool_proto),
+            std::move(logger)
+        );
+    }
+
+    template <typename Labels, typename ThreadPoolProto>
+    MultiMutableVamanaIndex(
+        Graph graph,
+        Data data,
+        label_type entry_point,
+        Dist distance_function,
+        const Labels& labels,
+        ThreadPoolProto threadpool_proto,
+        svs::logging::logger_ptr logger = svs::logging::get()
+    )
+        : distance_(std::move(distance_function)) {
+        std::vector<external_id_type> adds;
+        adds.reserve(labels.size());
+        prepare_added_id_by_label(labels, adds);
+
+        index_ = std::make_unique<ParentIndex>(
+            std::move(graph),
+            std::move(data),
+            label_to_external_.at(entry_point).at(0),
+            distance_,
+            std::move(adds),
+            std::move(threadpool_proto),
+            std::move(logger)
+        );
+    }
+
+    // change translator to label -> external_id
+    // create a transformed translator where external_id == internal_id
```

Comment:
```suggestion
    // Change translator to label -> external_id
    // Create a transformed translator where external_id == internal_id
```

Should these comments be below where this logic is implemented? Here we can have a short comment on what this function is doing